Source/WebKit/UIProcess/WebPermissionControllerProxy.cpp

```diff
@@ -57,4 +71,42 @@ void WebPermissionControllerProxy::query(const WebCore::ClientOrigin& clientOrig
     webPageProxy->queryPermission(clientOrigin, descriptor, WTFMove(completionHandler));
 }
 
+WebPageProxy* WebPermissionControllerProxy::mostReasonableWebPageProxy(const WebCore::SecurityOriginData& topOrigin, WebCore::PermissionQuerySource source)
+{
+    WebPageProxy* webPageProxy = nullptr;
+
+    auto findWebPageProxy = [&topOrigin, &webPageProxy] (auto* processes) {
+        if (!processes)
+            return; 
+
+        for (auto& process : *processes) {
+            for (auto* potentialWebPageProxy : getPtr(process)->pages()) {
+                if (WebCore::SecurityOriginData::fromURL(URL { potentialWebPageProxy->currentURL() }) != topOrigin)
+                    continue;
+                // The most reasonable webPageProxy is the newest one (the one with the greatest identifier).
+                if (webPageProxy && webPageProxy->identifier().toUInt64() > potentialWebPageProxy->identifier().toUInt64())
+                    continue;
+                webPageProxy = potentialWebPageProxy;
+            }
+        }
+    };
+
+    Vector<Ref<WebProcessProxy>> currentProcess;
```

Comment:
I think this can be written as:
Vector currentProcess { Ref { m_process } };

or similar. Pretty sure we can get this on 1 line instead of 2.

Reply:
Fixed!